### PR TITLE
Allow gene names with special characters by escaping inside backticks

### PR DESCRIPTION
**Pull request by chrismacklin**
Adds lexing, parsing, and tests for supplying gene names with weird characters (allows hyphen, comma, and parentheses) by enclosing the gene name in backticks.

### CODE REVIEW

File: src/GslCore/AstAlgorithms.fs [hunk 1]
```diff
@@ -26,6 +26,13 @@ type PrintState =
     insideAssembly: bool;
     quoteStrings: bool}
 
+/// Characters that appear in gene names that we need to escape with backticks
+/// when decompiling.
+let weirdGeneNameChars = [|'-'; ','; '('; ')'|]
```
Comment (daz10000) [inline src/GslCore/AstAlgorithms.fs]: Check out gene YIL154C.   I think it has a ' in its name.  It's officially IMP2'   

https://www.yeastgenome.org/locus/S000001416

File: src/GslCore/GslLexer.fsl [hunk 2]
```diff
@@ -19,6 +19,9 @@ type GslTokenizerMode =
 let num = ['0'-'9']+
 let intNum = '-'? num
 let ident = ['a'-'z' 'A'-'Z']['a'-'z' 'A'-'Z' '0'-'9' '_']*
+
+// Use backticks to escape gene literal names with odd characters.
+let geneName = '`' ['a'-'z' 'A'-'Z']['a'-'z' 'A'-'Z' '0'-'9' '_' '-' ',' '(' ')']* '`'
```
Comment (daz10000) [inline src/GslCore/GslLexer.fsl]: This probably needs to allow for IMP2' as well

**Review by daz10000 — CHANGES_REQUESTED** [review]
One other global suggestion - I'd throw a small MD file into docs/Content outlining the syntax.  We have shitty public docs outside a few chapters / papers,  so I have been trying to hold myself to standard of documenting anything new at least and eventually we will catch up.   At some point,  I or a summer intern will write a bunch more.

File: src/GslCore/AstAlgorithms.fs (hunk 1 above)
Comment (chrismacklin) [reply src/GslCore/AstAlgorithms.fs]: Thanks, I'll add that character!  I thought briefly about allowing a broad range of special characters here but decided to keep it restricted to the known problem cases.

**Comment by chrismacklin** [comment]
Added SYNTAX.md and added gene names with escapes as a start.

**Comment by chrismacklin** [comment]
Regression tests are green; bombs away!

**Pull request merged**